### ISSUE TITLE
strange list comprehension

### ISSUE THREAD

**Issue by LeiYangGH** [pytest-dev/pytest-metadata#63]
reading the source code, see [following snippet](https://github.com/pytest-dev/pytest-metadata/blob/master/src/pytest_metadata/plugin.py#L103):
which topologically equals to 
```python
for provider in [[1, 2], [3, 4]]:
    [
        print(var)
        for var in provider
        if var
    ]
```
I tried it and can print `1, 2, 3, 4` sequencely.
I think this can be rewritten as pure nested for loop or list comprehension.
why keep such strange syntax?

**Comment by BeyondEvil** [comment]
Happy to review a PR that simplifies the syntax!

**Comment by LeiYangGH** [comment]
i created https://github.com/pytest-dev/pytest-metadata/pull/64, could you review?

**Comment by BeyondEvil** [comment]
Approved!

This plugin really needs some love.

For instance, I'd really like to find a way where we don't have to manually maintain CI providers AND all the environment variables.

Or at the very least limit the maintenance burden.

This might be useful: https://adamj.eu/tech/2020/03/09/detect-if-your-tests-are-running-on-ci/

**Comment by BeyondEvil** [comment]
Fixed in #64
